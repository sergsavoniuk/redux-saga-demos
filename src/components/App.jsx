import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { push } from 'connected-react-router';

import Routes from 'components/Routes';
import Layout from 'components/Layout';
import AuthForm from 'components/AuthForm';
import Loader from 'components/Loader';
import { login, getIsAuthenticated } from 'redux/auth';

function getUsername() {
  return localStorage.getItem('username');
}

function App({ isAuthenticated, login, push }) {
  useEffect(() => {
    const username = getUsername();
    if (!isAuthenticated) {
      push('/auth');
      if (username != null) {
        login(username);
      }
    }
  }, [isAuthenticated]);

  let Component;
  if (!isAuthenticated) {
    Component = getUsername() != null ? <Loader /> : <AuthForm />;
  } else {
    Component = <Routes />;
  }

  return <Layout>{Component}</Layout>;
}

function mapStateToProps(state) {
  return {
    isAuthenticated: getIsAuthenticated(state),
  };
}

export default connect(
  mapStateToProps,
  { login, push },
)(App);
