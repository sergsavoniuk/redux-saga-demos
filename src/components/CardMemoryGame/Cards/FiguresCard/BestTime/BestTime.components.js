import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 50%;
`;

export const Row = styled.div`
  display: flex;
  justify-content: space-between;
  padding: 5px;
`;

export const Label = styled.span``;
export const Value = styled.span``;
