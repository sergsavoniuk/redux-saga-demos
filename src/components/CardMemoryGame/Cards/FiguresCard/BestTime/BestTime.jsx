import React from 'react';
import { format } from 'date-fns';

import {
  Wrapper,
  Row,
  Span as Label,
  Span as Value,
} from './BestTime.components';

const DEFAULT_PLACEHOLDER = '-:-';
const TIME_FORMAT = 'ss.SSS';

function BestTime({ casual, medium, hard }) {
  return (
    <Wrapper>
      <Row>
        <Label>Best Casual:</Label>
        <Value>
          {casual ? `${format(casual, TIME_FORMAT)}s` : DEFAULT_PLACEHOLDER}
        </Value>
      </Row>
      <Row>
        <Label>Best Medium:</Label>
        <Value>
          {medium ? `${format(medium, TIME_FORMAT)}s` : DEFAULT_PLACEHOLDER}
        </Value>
      </Row>
      <Row>
        <Label>Best Hard:</Label>
        <Value>
          {hard ? `${format(hard, TIME_FORMAT)}s` : DEFAULT_PLACEHOLDER}
        </Value>
      </Row>
    </Wrapper>
  );
}

export default BestTime;
