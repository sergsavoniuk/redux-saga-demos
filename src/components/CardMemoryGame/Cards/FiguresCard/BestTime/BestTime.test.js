import React from 'react';
import { shallow } from 'enzyme';

import BestTime, { DEFAULT_PLACEHOLDER } from './BestTime';
import { Row } from './BestTime.components';

export function getValueComponent(wrapper, rowIndex) {
  return wrapper
    .find(Row)
    .at(rowIndex)
    .children()
    .at(1); // Value component
}

describe('Testing BestTime component', () => {
  test('renders correctly without props - used default placeholder', () => {
    const wrapper = shallow(<BestTime />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Row)).toHaveLength(3);

    expect(getValueComponent(wrapper, 0).text()).toMatch(DEFAULT_PLACEHOLDER);
    expect(getValueComponent(wrapper, 1).text()).toMatch(DEFAULT_PLACEHOLDER);
    expect(getValueComponent(wrapper, 2).text()).toMatch(DEFAULT_PLACEHOLDER);
  });

  test('renders correctly with passed props', () => {
    const props = {
      casual: 10000,
      medium: 18000,
      hard: 27000,
    };

    const wrapper = shallow(<BestTime {...props} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Row)).toHaveLength(3);
  });
});
