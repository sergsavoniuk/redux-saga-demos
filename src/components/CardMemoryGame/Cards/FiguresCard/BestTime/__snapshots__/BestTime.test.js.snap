// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing BestTime component renders correctly with passed props 1`] = `
<styled.div>
  <styled.div>
    <styled.span>
      Best Casual:
    </styled.span>
    <styled.span>
      10.000s
    </styled.span>
  </styled.div>
  <styled.div>
    <styled.span>
      Best Medium:
    </styled.span>
    <styled.span>
      18.000s
    </styled.span>
  </styled.div>
  <styled.div>
    <styled.span>
      Best Hard:
    </styled.span>
    <styled.span>
      27.000s
    </styled.span>
  </styled.div>
</styled.div>
`;

exports[`Testing BestTime component renders correctly without props - used default placeholder 1`] = `
<styled.div>
  <styled.div>
    <styled.span>
      Best Casual:
    </styled.span>
    <styled.span>
      -:-
    </styled.span>
  </styled.div>
  <styled.div>
    <styled.span>
      Best Medium:
    </styled.span>
    <styled.span>
      -:-
    </styled.span>
  </styled.div>
  <styled.div>
    <styled.span>
      Best Hard:
    </styled.span>
    <styled.span>
      -:-
    </styled.span>
  </styled.div>
</styled.div>
`;
